Add info about managing packaged components (#316)
rancher/rke2-docs: docs/install/packaged_components.md

---
title: "Managing Packaged Components"
---

## Auto-Deploying Manifests (AddOns)

On server nodes, any file found in `/var/lib/rancher/rke2/server/manifests` will automatically be deployed to Kubernetes in a manner similar to `kubectl apply`, both on startup and when the file is changed on disk. Deleting files out of this directory will not delete the corresponding resources from the cluster.

Manifests are tracked as `AddOn` custom resources in the `kube-system` namespace. Any errors or warnings encountered when applying the manifest file may be seen by using `kubectl describe` on the corresponding `AddOn`, or by using `kubectl get event -n kube-system` to view all events for that namespace, including those from the deploy controller.

### Packaged Components

> RKE2 comes with a number of packaged components that are deployed as AddOns via the manifests directory, for example `coredns`, `metrics-server` or any of the cni plugins.

Reviewer: Give the correct addon name. Also lets not mention the CNI charts here, as the `--cni` flag should be used to manage those - we don't want people to use the --disable flag to poke at those directly.

```suggestion
RKE2 comes with a number of packaged components that are deployed as AddOns via the manifests directory, for example `rke2-coredns`, `rke2-metrics-server`, `rke2-ingress-nginx`, and so on.
```
